provider-assignments/src/java/org/sakaiproject/archiver/provider/AssignmentArchiver.java

```diff
@@ -0,0 +1,455 @@
+				try {
+					archiveAttachment(attachment, archiveId, siteId, toolId, submissionSubdirs);
+				} catch (ServerOverloadException | PermissionException | IdUnusedException | TypeException e) {
+					log.error("Error getting submission attachment for assignment: " + assignment.getTitle());
+				}
+			}
+			// get other data associated with this submission
+			if (submission.getTimeSubmitted() != null) {
+				SimpleSubmission submissionData = new SimpleSubmission(submission, assignment.isGroup());
+				this.archiverService.archiveContent(archiveId, siteId, toolId, Jsonifier.toJson(submissionData).getBytes(), "submission.json", submissionSubdirs);
+			}
+
+			// get the feedback, if this submission has been graded
+			if (submission.getGraded()) {
+				// get the attachments provided by the instructor when grading the submission
+				List<Reference> feedbackAttachments = submission.getFeedbackAttachments();
+				String feedbackSubdirs = getSubDirs(assignment, submission.getSubmitterId(), "feedback");
+				for (Reference attachment : feedbackAttachments) {
+					try {
+						archiveAttachment(attachment, archiveId, siteId, toolId, feedbackSubdirs);
+					} catch (ServerOverloadException | PermissionException | IdUnusedException | TypeException e) {
+						log.error("Error getting feedback attachment for assignment: " + assignment.getTitle());
+					}
+				}
+				
+				// get other data associated with this feedback
+				SimpleFeedback feedback = new SimpleFeedback(submission);
+				this.archiverService.archiveContent(archiveId, siteId, toolId, Jsonifier.toJson(feedback).getBytes(), "feedback.json", feedbackSubdirs);
+			}
+		}
+	}
+	
+	
+	/**
+	 * Helper method to archive an attachment
+	 * 
+	 * @param attachment
+	 * @param archiveId
+	 * @param siteId
+	 * @param toolId
+	 * @param title
+	 * @throws ServerOverloadException
+	 * @throws PermissionException
+	 * @throws IdUnusedException
+	 * @throws TypeException
+	 */
+	private void archiveAttachment(Reference attachment, String archiveId, String siteId, String toolId, String subdir) throws ServerOverloadException, PermissionException, IdUnusedException, TypeException {
+		byte[] attachmentBytes = this.contentHostingService.getResource(attachment.getId()).getContent();
+		this.archiverService.archiveContent(archiveId, siteId, toolId, attachmentBytes, 
+				attachment.getProperties().getPropertyFormatted(attachment.getProperties().getNamePropDisplayName()), subdir);
+	}
+
+	/**
+	 * Get the subdirectory structure for a submission or feedback item
+	 * @param assignmentTitle
+	 * @param submitterId
+	 * @param folderName
+	 * @return subdirectory string
+	 */
+	private String getSubDirs(Assignment assignment, String submitterId, String folderName) {
+
+		// Get the user associated with this submitterId
+		User user = getUser(submitterId);
+		if (user != null) {
+			return assignment.getTitle() + "/submissions/" + user.getLastName() + ", " + user.getFirstName() + "/" + folderName;
```

Comment:
user.getSortName() instead of the concatentation, deals with different languages etc.